app/shared/Shared/Services/AzureDocumentService.cs

```diff
@@ -0,0 +1,123 @@
+﻿using Azure;
+using Azure.Search.Documents;
+using Azure.Search.Documents.Models;
+using Shared.Models;
+
+public interface IDocumentService
+{
+    Task<SupportingContentRecord[]> QueryDocumentsAsync(
+               string? query = null,
+               float[]? embedding = null,
+               RequestOverrides? overrides = null,
+               CancellationToken cancellationToken = default);
+}
+
+public class AzureDocumentService(SearchClient searchClient) : IDocumentService
+{
+    public async Task<SupportingContentRecord[]> QueryDocumentsAsync(
+        string? query = null,
+        float[]? embedding = null,
+        RequestOverrides? overrides = null,
+        CancellationToken cancellationToken = default)
+    {
+        if (query is null && embedding is null)
+        {
+            throw new ArgumentException("Either query or embedding must be provided");
+        }
+
+        var documentContents = string.Empty;
+        var top = overrides?.Top ?? 3;
+        var exclude_category = overrides?.ExcludeCategory;
+        var filter = exclude_category == null ? string.Empty : $"category ne '{exclude_category}'";
+        var useSemanticRanker = overrides?.SemanticRanker ?? false;
+        var useSemanticCaptions = overrides?.SemanticCaptions ?? false;
+
+        SearchOptions searchOptions = useSemanticRanker
+            ? new SearchOptions
+            {
+                Filter = filter,
+                QueryType = SearchQueryType.Semantic,
+                SemanticSearch = new()
+                {
+                    SemanticConfigurationName = "default",
+                    QueryCaption = new(useSemanticCaptions
+                        ? QueryCaptionType.Extractive
+                        : QueryCaptionType.None),
+                },
+                // TODO: Find if these options are assignable
+                //QueryLanguage = "en-us",
+                //QuerySpeller = "lexicon",
+                Size = top,
+            }
+            : new SearchOptions
+            {
+                Filter = filter,
+                Size = top,
+            };
+
+        if (embedding != null && overrides?.RetrievalMode != "Text")
```

Comment:
Why is the `RetrievalMode` a `string` type? Might be safer to instead refactor to defining and sharing an `enum`.